File: app/src/release/kotlin/com/example/newsapp/BuildTypeApplication.java
package com.example.newsapp;

import android.app.Application;
import android.util.Log;

import com.crashlytics.android.Crashlytics;

import io.fabric.sdk.android.Fabric;
import timber.log.Timber;

public class BuildTypeApplication extends Application {
    @Override
    public void onCreate() {
        super.onCreate();
        Timber.plant(new Timber.Tree() {
            @Override
            protected void log(int priority, String tag, String message, Throwable throwable) {
                if (priority >= Log.INFO && Fabric.isInitialized()) {
                    Crashlytics.log(priority, tag, message);
                    if (throwable != null) {
                        Crashlytics.logException(throwable);
                    }
                }
            }
        });
    }
}

File: app/src/debug/kotlin/com/example/newsapp/BuildTypeApplication.kt
package com.example.newsapp

import android.app.Application
import android.view.View

import com.facebook.stetho.Stetho
import com.facebook.stetho.timber.StethoTree
import com.squareup.leakcanary.LeakCanary
import com.squareup.leakcanary.RefWatcher

import timber.log.Timber

open class BuildTypeApplication : Application() {
    private lateinit var refWatcher: RefWatcher

    override fun onCreate() {
        super.onCreate()
        Stetho.initialize(Stetho.newInitializerBuilder(this)
                .enableDumpapp(Stetho.defaultDumperPluginsProvider(this))
                .enableWebKitInspector(Stetho.defaultInspectorModulesProvider(this))
                .build())
        Timber.plant(Timber.DebugTree())
        Timber.plant(StethoTree())

        refWatcher = LeakCanary.install(this)
    }

    companion object {
        fun watchView(view: View) {
            (view.context.applicationContext as BuildTypeApplication).refWatcher.watch(view)
            Timber.d("LeakCanary watching object reference: %s", view)
        }
    }
}
